runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -817,6 +798,41 @@ public synchronized CorfuRuntime connect() {
         return this;
     }
 
+    public BaseSenderClient getBaseClient(Layout layout, String endpoint) {
+        return new BaseSenderClient(getRouter(endpoint), layout.getEpoch());
+    }
+
+    public LayoutSenderClient getLayoutClient(Layout layout, String endpoint) {
+        return new LayoutSenderClient(getRouter(endpoint), layout.getEpoch());
+    }
+
+    public SequencerSenderClient getPrimarySequencerClient(Layout layout) {
+        return getSequencerClient(layout, 0);
+    }
+
+    public SequencerSenderClient getSequencerClient(Layout layout, int index) {
+        return new SequencerSenderClient(getRouter(layout.getSequencers().get(index)),
```

Comment:
This new object thing might be a little bit of a performance hit if we have to do it for every message.

Reply:
Changed it to fetch a client from a cache in the runtime.